WORKSPACE

```diff
@@ -304,5 +306,72 @@ rbe_autoconfig(
 
 load("@build_bazel_integration_testing//tools:repositories.bzl", "bazel_binaries")
 
-#depend on the Bazel binaries, also accepts an array of versions
-bazel_binaries()
+# Depend on the Bazel binaries
+bazel_binaries(versions = ["0.28.1"])
+
+#
+# Support creating Docker images for our node apps #
+# Not used but needed for @e2e_angular_bazel_example//:bazel_integration_test_files target
+#
+
+http_archive(
+    name = "io_bazel_rules_docker",
+    sha256 = "aed1c249d4ec8f703edddf35cbe9dfaca0b5f5ea6e4cd9e83e99f3b0d1136c3d",
+    strip_prefix = "rules_docker-0.7.0",
+    urls = ["https://github.com/bazelbuild/rules_docker/archive/v0.7.0.tar.gz"],
+)
+
+load("@io_bazel_rules_docker//nodejs:image.bzl", nodejs_image_repos = "repositories")
+
+nodejs_image_repos()
+
+#
+# Kubernetes setup, for deployment to Google Cloud
+# Not used but needed for @e2e_angular_bazel_example//:bazel_integration_test_files target
+#
+
+git_repository(
+    name = "io_bazel_rules_k8s",
+    commit = "36ae5b534cc51ab0815c9bc723760469a9f7175c",
+    remote = "https://github.com/bazelbuild/rules_k8s.git",
+    shallow_since = "1545317854 -0500",
+)
+
+load("@io_bazel_rules_k8s//k8s:k8s.bzl", "k8s_defaults", "k8s_repositories")
+
+k8s_repositories()
+
+k8s_defaults(
+    # This creates a rule called "k8s_deploy" that we can call later
+    name = "k8s_deploy",
+    # This is the name of the cluster as it appears in:
+    #   kubectl config view --minify -o=jsonpath='{.contexts[0].context.cluster}'
+    cluster = "_".join([
+        "gke",
+        "internal-200822",
+        "us-west1-a",
+        "angular-bazel-example",
+    ]),
+    kind = "deployment",
+)
+
+#
+# Setup bazel_integration_test repositories
+#
+
+local_repository(
+    name = "e2e_jasmine",
+    path = "e2e/jasmine",
+)
+
+git_repository_under_test(
```

Comment:
curious: would this be any easier if we moved the code now? Looks like you had to do some mildly crazy stuff to get this working

Reply:
yes. it would probably be easier in this case. although the mildly crazy stuff could still come in handle in the future for running integration tests against other repos.

Reply:
In the external repo case I think I could get it to work with less crazy stuff if the external repo was not in runfiles. In that case, test_runner would just copy all the files from the external repo but these would not be inputs to the test so if the external repo was updated the test would not automatically re-run. At least in the angular-bazel-example case this may be better as adding `workspace_files = "@e2e_angular_bazel_example//:bazel_integration_test_files",` meant having to add `io_bazel_rules_docker`, `io_bazel_rules_k8s` and mock `npm_angular_bazel` repositories to the root WORKSPACE even tho they are not used since `@e2e_angular_bazel_example//:BUILD.bazel` references those repos.

Reply:
It's actually really problematic to have internal and external details in BUILD files in external repositories. I think Bazel could improve this by either adding built-in build file fencing or maybe even better allow a parallel `BUILD.private` file which is meant for internal only build targets so that the repositories references and used for those targets don't leak into the public/external BUILD file targets.